### ISSUE TITLE
Displayed picture jumps to beginning in file manager when rating pictures out of view

### ISSUE THREAD

**Issue by MeisterKaos** [darktable-org/darktable#7774]
This is similar to #6566 and probably related. I discussed it there with @pahl34 first and we decided to open a new report since that one is closed and the issue now was observed in file manager instead of culling (where it seems to be fixed).

Issue: When rating images in 'file manager' in a way, that they should afterwards disappear fromm the view (because the star rating is outside the set 'view'), the strip jumps back to the first image instead of the next. This is also reproduceable through collapsing images (see below). I recorded the issue (both ways) in a screengrab (attached), which is hopefully helpful.

**To Reproduce**
1. Load a collection of pictures shot in JPG and RAW. which are all rated 1 star.
2. Open file manager mode
3. Set the rating 'view' to >=1 star.
4. Collapse grouped images.
5. Set the file manager slider in the bottom to 1 image displayed.
6. Scroll a bit into the collection.
7. Rate a picture 0 star by hitting '1' shortcut (rating an image one star twice to zero out the rating)
--> Filmstrip jumps back to the start (first picture)

**Expected behavior**
After rating the picture out of star-selection, the NEXT picture in the selection should appear (not the first).

**Platform (please complete the following information):**
 - Darktable Version: Self built windows version of 3.4.x branch
 - OS: Windows 10

**Additional context**
- I can also reproduce it with different rating/view setting, e.g. view>=2 and rating an image 1 star. 
- The jumping does NOT occur when the rating is done by clicking on the star-rating below the image.
- @pahl34 could reproduce it on Debian 10 on master 3.5.0\~git480.4f8915dc3 as well as 3.5.0\~git294.9f83abb98
- There is an alternative way to reproduce the jumping (which is probably related but less annoying):

1. Load a collection of pictures shot in JPG and RAW.
2. Open file manager mode and set the slider to 1 image.
3. Expand grouped images (RAW are group header).
4. Scroll a bit into the collection.
5. Select a JPG (click on it).
6. Collapse grouped images.
    --> Filmstrip jumps back to the start (first picture)

https://user-images.githubusercontent.com/69791606/104124941-fdc6fe80-5353-11eb-81fd-99a538a56d5a.mp4

**Comment by pahl34** [comment]
@MeisterKaos: Thanks for opening the issue!
Minor remark: To avoid the crossed out text in the build label, you could escape the tilde (\~). Right now it could be a bit confusing that the labels of versions in which the problem is actually reproducible are shown crossed out.

@AlicVB: This issue describes a very similar problem to #6566 that remains after your fix. There are workarounds (e. g. use culling instead of file manager), but I thought it might be interesting for you.

**Comment by AlicVB** [comment]
at first sight, the problem seems closely related to the fact that in filemanager with zoom == 1, if the image disappear, we can't rely on anything to find the next image. In culling, we rely on the filmstrip (even if it's hidden), but here, we have no reference.

One solution could be to "backup" the memory.collected_image, so we can rely on something...
I'll have a look asap, but this may take some time, as I'm currently quite busy with other issues/features...

**Comment by MeisterKaos** [comment]
Hmm, I'm pretty sure it did work in 3.0 but of course I have no idea what changed under the hood for 3.2 that did introduce these problems. Naively, I would expect the easiest solution is to just do the same thing as when finding the next picture while scrolling? Obviously this would have to be done before removing the previous picture.
But thanks for looking into it! I'll manage with the workaround of using culling until then...

**Comment by AlicVB** [comment]
@MeisterKaos : I've finally started to work on that. In fact there's 2 issues in one here : 
1. the one described in your first step by step. I've not managed to reproduce, but I think I've understand what happen : for some reason, you have jpeg first and raw second (in term of image id) here it's always in reverse order. That's not a problem, but it expose a bug in the routine used to detect the first untouched image after the one modified. PR #7951 should fix it
2. the one described in your second setp by step. I reproduce this one. The problem here is that with filemanager and zoom == 1, we have nothing to rely on to find the right position. The only way would be to somehow save the previous collection. I've tried, but that add lot of complexity in the code and some performance penalties. So I would prefer to not fix this one. After all this is a very corner case, which should not really happen often in workflows...
